### PR TITLE
Users/shreyas r msft/hydra single agent rerun flow

### PR DESCRIPTION
**Pull request by ShreyasRmsft**

### CODE REVIEW

File: Tasks/VsTestV2/inputparser.ts [hunk 1]
```diff
@@ -25,20 +30,37 @@ export function getDistributedTestConfigurations() : idc.InputDataContract {
...
+    inputDataContract = getExecutionSettings(inputDataContract);
 
     inputDataContract.TeamProject = tl.getVariable('System.TeamProject');
     inputDataContract.CollectionUri = tl.getVariable('System.TeamFoundationCollectionUri');
-    inputDataContract.AccessToken = tl.getEndpointAuthorization('SystemVssConnection', true).parameters['AccessToken'];
+    inputDataContract.AccessToken = tl.getEndpointAuthorization('SystemVssConnection', true).parameters.AccessToken;
     inputDataContract.AgentName = tl.getVariable('Agent.MachineName') + '-' + tl.getVariable('Agent.Name') + '-' + tl.getVariable('Agent.Id');
     inputDataContract.RunIdentifier = getRunIdentifier();
 
+    logWarningForWER(tl.getBoolInput('uiTests'));
```
Comment (prawalagarwal) [inline Tasks/VsTestV2/inputparser.ts]: Add to telemetry

File: Tasks/VsTestV2/inputparser.ts [hunk 2]
```diff
@@ -135,6 +160,8 @@ function getTestReportingSettings(inputDataContract : idc.InputDataContract) : i
         inputDataContract.TestReportingSettings.TestRunTitle = `TestRun_${definitionName}_${buildOrReleaseName}`;
     }
 
+    //inputDataContract.TestReportingSettings.TestResultDirectory = read from xml here? or in C# layer?
```
Comment (prawalagarwal) [inline Tasks/VsTestV2/inputparser.ts]: remove

File: Tasks/VsTestV2/nondistributedtest.ts [hunk 3]
```diff
@@ -0,0 +1,162 @@
...
+    if (utils.Helper.isDebugEnabled()) {
+        utils.Helper.uploadFile(inputFilePath);
+    }
+
+    dtaExecutionHostTool.arg(['--inputFile', inputFilePath]);
+
+    utils.Helper.addToProcessEnvVars(envVars, 'DTA.AccessToken', tl.getEndpointAuthorization('SystemVssConnection', true).parameters.AccessToken);
+
+    // hydra: See which of these are required in C# layer. Do we want this for telemetry??
+    // utils.Helper.addToProcessEnvVars(envVars, 'DTA.AgentVersion', tl.getVariable('AGENT.VERSION'));
```
Comment (prawalagarwal) [inline Tasks/VsTestV2/nondistributedtest.ts]: add a task to fire telemetry from single agent flow

File: Tasks/VsTestV2/nondistributedtest.ts [hunk 4]
```diff
@@ -0,0 +1,162 @@
...
+        return 1;
+    }
+}
+
+function getTestAssemblies(): string[] {
+    tl.debug('Searching for test assemblies in: ' + inputDataContract.TestSelectionSettings.SearchFolder);
+    return tl.findMatch(inputDataContract.TestSelectionSettings.SearchFolder, sourceFilter);
+}
+
+function isTestAdapterPresent(rootDirectory: string): boolean {
```
Comment (prawalagarwal) [inline Tasks/VsTestV2/nondistributedtest.ts]: remove this,

File: Tasks/VsTestV2/nondistributedtest.ts [hunk 5]
```diff
@@ -0,0 +1,162 @@
...
+    const adapterFiles = tl.findMatch(rootDirectory, '**\\*TestAdapter.dll');
+
+    if (adapterFiles && adapterFiles.length !== 0) {
+        return true;
+    }
+    return false;
+}
+
+function isEmptyResponseFile(responseFile: string): boolean {
```
Comment (prawalagarwal) [inline Tasks/VsTestV2/nondistributedtest.ts]: remove

File: Tasks/VsTestV2/inputparser.ts (hunk 1 above)
Comment (ShreyasRmsft) [reply Tasks/VsTestV2/inputparser.ts]: done

File: Tasks/VsTestV2/nondistributedtest.ts (hunk 3 above)
Comment (ShreyasRmsft) [reply Tasks/VsTestV2/nondistributedtest.ts]: done

**Pull request merged**